composables/useModal.js

```diff
@@ -0,0 +1,53 @@
+import { useRouter } from 'vue-router'
+import { onUnmounted } from 'vue'
+
+export function useModal(emit) {
```

Comment:
Now we have both a composable and a mixin with the same function.  Composable is better, so we need to migrate to that in the places where we currently use the mixin.  It doesn't need to in this PR - please create another issue for that.

Reply:
Done.